Fabric.Authorization.AccessControl/src/app/modules/access-control/member/member.component.ts

```diff
@@ -0,0 +1,223 @@
+    return this.userService
+      .getUser(
+        this.configService.identityProvider,
+        subjectId
+      )
+      .mergeMap((userResult: IUser) => {
+        return Observable.of(userResult);
+      })
+      .catch(err => {
+        if (err.statusCode === 404) {
+          return this
+            .userService
+            .createUser(user);
+        }
+
+        return Observable.throw(err.message);
+      })
+      .mergeMap((newUser: IUser) => {
+        return this.userService.getUserRoles(user.identityProvider, user.subjectId);
+      })
+      .mergeMap((userRoles: IRole[]) => {
+        const rolesToAdd = selectedRoles.filter(userRole => !userRoles.some(selectedRole => userRole.id === selectedRole.id));
+        const rolesToDelete = userRoles.filter(userRole => !selectedRoles.some(selectedRole => userRole.id === selectedRole.id));
+        return Observable.zip(
+          this.userService
+            .addRolesToUser(
+              user.identityProvider,
+              user.subjectId,
+              rolesToAdd
+            ),
+          this.userService
+            .removeRolesFromUser(
+              user.identityProvider,
+              user.subjectId,
+              rolesToDelete
+            ));
+      });
+  }
+
+  private saveGroup(subjectId: string, selectedRoles: IRoleModel[]): Observable<any> {
+    const group: IGroup = { groupName: subjectId, groupSource: '' };
+    return this.groupService
+      .getGroup(group.groupName)
+      .mergeMap((groupResult: IGroup) => {
+        return Observable.of(groupResult);
+      })
+      .catch(err => {
+        if (err.statusCode === 404) {
+          return this
+            .groupService
+            .createGroup(group);
+        }
+
+        return Observable.throw(err.message);
+      })
+      .mergeMap((newGroup: IGroup) => {
+        return this.groupService.getGroupRoles(group.groupName,
+          this.configService.grain,
+          this.configService.securableItem);
+      })
+      .mergeMap((groupRoles: IRole[]) => {
+        const rolesToAdd = selectedRoles.filter(userRole => !groupRoles.some(selectedRole => userRole.id === selectedRole.id));
+        const rolesToDelete = groupRoles.filter(userRole => !selectedRoles.some(selectedRole => userRole.id === selectedRole.id));
+        return Observable.zip(
+          this.groupService.addRolesToGroup(group.groupName, rolesToAdd),
+          this.groupService.removeRolesFromGroup(group.groupName, rolesToDelete));
+      });
+  }
+
+  private bindExistingRoles(subjectId: string, principalType: string): Observable<any> {
+    if (!this.subjectId) {
+      return Observable.of(undefined);
+    }
+    const roleObservable = principalType === 'user' ?
+      this.userService.getUserRoles(this.configService.identityProvider, subjectId) :
+      this.groupService.getGroupRoles(subjectId, this.configService.grain, this.configService.securableItem);
+
+    return roleObservable.map((existingRoles: IRole[]) => {
```

Comment:
when calling `Observable.map`, shouldn't you be returning something?

Reply:
good catch!

Reply:
I think this will return an Observable<Observable<IRole>>.  Is this what you intended?